evalml/data_checks/default_data_checks.py

```diff
@@ -47,12 +49,13 @@ def __init__(self, problem_type, objective, n_splits=3):
             ProblemTypes.TIME_SERIES_REGRESSION,
         ]:
             super().__init__(
-                self._DEFAULT_DATA_CHECK_CLASSES,
+                self._DEFAULT_DATA_CHECK_CLASSES + [TargetDistributionDataCheck],
                 data_check_params={
                     "InvalidTargetDataCheck": {
                         "problem_type": problem_type,
                         "objective": objective,
-                    }
+                    },
+                    "TargetDistributionDataCheck": {"problem_type": problem_type},
```

Comment:
So these were the reasons I felt it was necessary:

- I had originally added it in case open source users try leveraging it against their data outside of `DefaultDataChecks`. A binary or multiclass problem that consists of a few unique numbers in the target data could actually be interpreted by the `TargetDistributionDataCheck` as having a lognormal distribution although that data check message wouldn't make much sense for non regression problems, and a user without much data science experience might not know what to do with it.
- I don't think most DataChecks need `problem_type` in their implementation, so we leave it out. However I do feel like `ClassImbalanceDataCheck` not having it is more of an argument in favour of including it for `DataChecks` that rely on `problem_type` like `UniquenessDataCheck` rather than against including it here.

I don't mind removing it at all, since adding it here makes me feel like we should add it for any `DataChecks` that rely on the exclusion of a problem type. What do you think?

Reply:
@ParthivNaresh This makes sense. I think this does set a new precedent of having to add the `problem_type` arg to `DataChecks` that rely on specific problem types, but that isn't necessarily a bad thing. If this seems right to you, we can move forward with it!